app/src/test/kotlin/fr/free/nrw/commons/nearby/NearbyParentFragmentPresenterTest.kt

```diff
@@ -0,0 +1,320 @@
+package fr.free.nrw.commons.nearby
+
+import com.nhaarman.mockitokotlin2.*
+import fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsDao
+import fr.free.nrw.commons.location.LatLng
+import fr.free.nrw.commons.location.LocationServiceManager
+import fr.free.nrw.commons.nearby.contract.NearbyParentFragmentContract
+import fr.free.nrw.commons.nearby.presenter.NearbyParentFragmentPresenter
+import org.junit.Before
+import org.junit.Test
+import org.junit.Assert.*
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+
+
+/**
+ * The unit test class for NearbyParentFragmentPresenter
+ */
+class NearbyParentFragmentPresenterTest {
+    @Mock
+    internal lateinit var nearbyParentFragmentView: NearbyParentFragmentContract.View
+    @Mock
+    internal lateinit var bookmarkLocationsDao: BookmarkLocationsDao
+    @Mock
+    internal lateinit var latestLocation: LatLng
+    @Mock
+    internal lateinit var cameraTarget: LatLng
+    @Mock
+    internal lateinit var selectedLabels: List<Label>
+
+    private lateinit var nearbyPresenter: NearbyParentFragmentPresenter
+    private lateinit var latestLocationSpy: LatLng
+    private lateinit var mapboxCameraTarget: com.mapbox.mapboxsdk.geometry.LatLng
+
+    /**
+     * initial setup
+     */
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        nearbyPresenter = Mockito.spy(NearbyParentFragmentPresenter(bookmarkLocationsDao))
+        nearbyPresenter.attachView(nearbyParentFragmentView)
+    }
+
+    /**
+     * Tests nearby operations are initialized
+     */
+    @Test
+    fun testInitializeNearbyMapOperations() {
```

Comment:
I know we add javadocs to classes but to tests? that seems extreme to me especially when you can have kotlin test functions enclosed in backticks to be named arbitrarily. For example the name of this test could be 
````
fun `Tests nearby operations are initialized`()
````

Reply:
I actually followed the style from other .kt files in our repository. If you see ContributionsPresenterTest.kt a similar style is followed.
```
    /**
     * Test fetch contributions
     */
    @Test
    fun testFetchContributions(){
```
Please let me know if you still think that the style should be changed:)

Reply:
@misaochan @ashishkumar0207 @maskaravivek thoughts?

One of Javadocs main usefulness is when trying to use the methods of a class you get the quick documentation without having to visit the class. Nobody should be directly using test methods.

I would say if the title of a test can be written clearly then repeating it in a javadoc is just busy work. For sufficiently complicated and/or confusing tests I would support commenting them at discretion or request.

Reply:
I think adding it for all of them can prevent relativity of complexity. If we don't add them for all, we can find out some complex tests written without tests. So I see no harm adding them, with as much as details we could add.

Reply:
I'm in support of continuing to require that they be added, mostly because it creates a lot of grey area otherwise (it may be difficult for contributors to agree on which tests are "complex enough to require javadocs" and which ones are not).

Reply:
To me tests are already documentation, documenting them again feels redundant. If ye are happy to proceed though then thats that.